scripts/src/java/org/oppia/android/scripts/RegexPatternValidationCheck.kt

```diff
@@ -0,0 +1,227 @@
+    @JvmStatic
+    fun main(vararg args: String) {
+      // path of the repo to be analyzed.
+      val repoPath = args[0] + "/"
+
+      // a list of all allowed directories in the repo to be analyzed.
+      // args[0] is the repoPath, the allowed directories are specified
+      // after it. Hence, we have to start from the 1st index.
+      val allowedDirectories = args.drop(1)
+
+      // a list of all files in the repo to be analyzed.
+      val searchFiles = RepoFile.collectSearchFiles(
+        repoPath,
+        allowedDirectories
+      )
+
+      // class object which is needed to acess the helper methods
+      val regexPatternValidationCheck: RegexPatternValidationCheck =
+        RegexPatternValidationCheck()
+
+      // check if the repo has any filename failure.
+      val hasFilenameCheckFailure = regexPatternValidationCheck.retrieveFilenameChecks()
+        .fold(initial = false) { isFailing, filenameCheck ->
+          val checkFailed = regexPatternValidationCheck.checkProhibitedFileNamePattern(
+            repoPath,
+            searchFiles,
+            filenameCheck,
+          )
+          isFailing || checkFailed
+        }
+
+      // check if the repo has any file content failure.
+      val hasFileContentCheckFailure = regexPatternValidationCheck.retrieveFileContentChecks()
+        .fold(initial = false) { isFailing, fileContentCheck ->
+          val checkFailed = regexPatternValidationCheck.checkProhibitedContent(
+            repoPath,
+            searchFiles,
+            fileContentCheck,
+          )
+          isFailing || checkFailed
+        }
+
+      if (hasFilenameCheckFailure || hasFileContentCheckFailure) {
+        throw Exception(ScriptResultConstants.REGEX_CHECKS_FAILED)
+      } else {
+        println(ScriptResultConstants.REGEX_CHECKS_PASSED)
+      }
+    }
+  }
+
+  /**
+   * Retrieves all filename checks.
+   *
+   * @return a list of all the FilenameChecks
+   */
+  private fun retrieveFilenameChecks(): List<FilenameCheck> {
+    return getProto(
+      "filename_pattern_validation_checks.pb",
+      FilenameChecks.getDefaultInstance()
+    ).getFilenameChecksList()
+  }
+
+  /**
+   * Retrieves all file content checks.
+   *
+   * @return a list of all the FileContentChecks
+   */
+  private fun retrieveFileContentChecks(): List<FileContentCheck> {
+    return getProto(
+      "file_content_validation_checks.pb",
+      FileContentChecks.getDefaultInstance()
+    ).getFileContentChecksList()
+  }
+
+  /**
+   * Helper function to parse the textproto file to a proto class.
+   *
+   * @param textProtoFileName name of the textproto file to be parsed
+   * @param proto instance of the proto class
+   * @return proto class from the parsed textproto file
+   */
+  private fun <T : MessageLite> getProto(textProtoFileName: String, proto: T): T {
+    val protoBinaryFile = File("scripts/assets/$textProtoFileName")
+    val builder = proto.newBuilderForType()
+
+    @Suppress("UNCHECKED_CAST")
```

Comment:
```suggestion
    // This cast is type-safe since proto guarantees type consistency from mergeFrom(), and this method is bounded by the generic type T.
    @Suppress("UNCHECKED_CAST")
```

Might need line wrapping. This comment is useful--whenever we suppress legitimate warnings, we should explain why that's fine to do.

Reply:
Done.